src/main/kotlin/Shovel.kt

```diff
@@ -34,21 +36,27 @@ fun shovelAsync(consumer: KafkaConsumer<ByteArray, ByteArray>, hbase: HbaseClien
                 val records = consumer.poll(pollTimeout)
 
                 if (records.count() > 0) {
-                    logger.info("Processing records", "record_count", records.count().toString())
-                    for (record in records) {
-                        //TODO: Implement saving record to the metadata store database before sending to hbase in case hbase loses it
-                        processor.processRecord(record, hbase, parser)
-                        offsets[record.topic()] = mutableMapOf(
-                            "offset" to "${record.offset()}",
-                            "partition" to "${record.partition()}"
-                        )
-                        val set =
-                            if (usedPartitions.containsKey(record.topic())) usedPartitions[record.topic()] else mutableSetOf()
-                        set?.add(record.partition())
-                        usedPartitions[record.topic()] = set!!
+                    val then = Date().time
+                    try {
+                        logger.info("Processing records", "record_count", records.count().toString())
+                        for (record in records) {
+                            //TODO: Implement saving record to the metadata store database before sending to hbase in case hbase loses it
+                            processor.processRecord(record, hbase, parser)
+                            offsets[record.topic()] = mutableMapOf(
+                                "offset" to "${record.offset()}",
+                                "partition" to "${record.partition()}"
+                            )
+                            val set =
+                                if (usedPartitions.containsKey(record.topic())) usedPartitions[record.topic()] else mutableSetOf()
+                            set?.add(record.partition())
+                            usedPartitions[record.topic()] = set!!
+                        }
+                        logger.info("Committing offset")
+                        consumer.commitSync()
+                    } finally {
+                        val now = Date().time
```

Comment:
This isn't a batch time though, it's an individual record

Reply:
Batch times. This will give us the stats for how long batches take for the k2hb timeouts.